Refactor create_image and create_secrets

dmendiza opened the pull request:
This patch changes the create_image tasks to use the Linux Minimal client as provided by Thales.

Adding binaries is not necessary because they are included in recent versions of the minimal client.

dmendiza committed:
Refactore create_image

This patch changes the create_image tasks to use the Linux Minimal
client as provided by Thales.

Adding binaries is not necessary because they are included in recent
versions of the minimal client.

vakwetu commented on README.md:
```diff
@@ -1,35 +1,37 @@
+    When using more than one HSM appliance, this file will be a concatenation of all the server certificates.
   * The client certificate and key made available at luna_client_cert_src.  The files are expected
-    to be of the form "(client_ip)".pem and "(client_ip)"Key.pem
+    to be of the form $(CLIENT_NAME).pem and $(CLIENT_NAME)Key.pem
```
We've made this change here, but have not changed the parameter - which is still called client_ip.
This parameter probably needs some explanation as well.

dmendiza replied:
Yeah, sorry I started to change this when I updated the doc but haven't gotten to the actual variable yet.  I mainly want to rename this because `client_ip` implies that it has to be an IP, but this could be either a hostname or an IP.  I was thinking `client_host` at first since host can be either a hostname or IP.

vakwetu commented on defaults/main.yml:
```diff
@@ -24,12 +24,12 @@ cleanup: false
-luna_binaries_src: "file:///opt/luna/bin/"
-luna_server_cert_src: "file:///opt/luna/cert/server/cacert.pem"
+luna_minclient_src: "file:///opt/luna/LunaClient-Minimal-10.7.2-16.x86_64.tar"
+luna_minclient_version: "10.7.2-16"
```
Do we plan to do anything with this parameter?  Otherwise, if this is just informational, it should probably go into the README instead.

dmendiza replied:
I am actually using it in the test playbook that calls into this role, but I did not update the defaults that need it.  I'll update them in the next patch.  Long story short, the minimal client tarballs that Thales distributes extract to a parent directory with this pattern: `LunaClient-Minimal-$(VERSION)-$(BUILD)/*.*` and `luna_minclient_version` here corresponds to the `$(VERSION)-$(BUILD)` part. 

I considired grepping this out of the filename, but that is not guaranteed to work every time because the filename of the tarball could be changed arbitrarily.

vakwetu commented on defaults/main.yml:
```diff
@@ -24,12 +24,12 @@ cleanup: false
 luna_client_cert_src: "file:///opt/luna/cert/client/"
 chrystoki_conf_src: "file:///opt/luna/Chrystoki.conf"
-server_ca_file: "cacert.pem"
+server_ca_file: "CAFile.pem"
```
Its OK to change this of course, but presumably whatever is there needs to match what is going to be in Chrystoki.conf.  I think we probably need some details in the README describing how the locations of the certs in this file need to be filled out.  I wonder if one can use relative paths, then we wouldn't have to worry about where the top level is.

dmendiza replied:
I'm only changing filenames to match the names used by default in the HSM client software.  In this case, both `vtl` and the Chrystoki.conf template included with the minimal client use `CAFile.pem`

I agree, it would be nicer to just worry about file names in most of these.  Maybe having a variable for a top level of the prereqs (inputs?) would be useful here.

dmendiza committed:
Refactor create_secrets tasks

This patch updates a few variables to make the role more consistent with
Luna HSM Client software defaults.

vakwetu commented on README.md:
```diff
@@ -1,78 +1,102 @@
+        luna_minclient_src: "file:///opt/luna/LunaClient-Minimal-{{ luna_minclient_version }}.x86_64.tar"
+       tasks:
+       - name: Create new barbican images with the Luna Minimal Client
+         ansible.builtin.includ_role:
```
NIT: typo - include_role

vakwetu reviewed (approved):
Minor nit on the README, but otherwise looks good.  I like the table on the README.

We still need some description that tells users a little how to create a Crystoki.conf - maybe at east letting them know how to reference the relevant certs.  But we can defer that to a future commit.

I'm wondering if we want to be more generic on the client-data.  That is, rather than having to specify the client_name and crystoki.conf location, simply provide a client_data_dir location, and scoop any files in there into a secret.  We'd of course have to tell them the types of things we'd want in there for example, but this allows them to add other files if needed.  We can defer this too to a new commit.

vakwetu committed:
Update README.md
